src/align.py

```diff
@@ -69,6 +71,45 @@ def make_modified_TarInfo(archive, target_dir=''):
     return members
 
 
+def choices(population, k):
+    result = []
+    for i in range(k):
+        result.append(random.choice(population))
+    return result
+
+
+def get_tmp_file_name(extension=".fastq.gz"):
+    """
+    Returns a string that contains 20 random lowercase letters followed
+    by extension. Used to guarantee that the name of the merged filename
+    will not collide with anything.
+    """
+    while True:
+        tmp_name = ''.join(choices('abcdefghijklmnopqrstuvwxyz', k=20)) + extension
+        if pathlib.Path(tmp_name).exists():
+            continue
+        else:
+            return tmp_name
+
+
+def concatenate_files(input_files):
+    """Merge list of files into one.
+    Args: list of paths.
+    Returns: pathlib.Path to the concatenated file.
+    Side effect: creates the concatenated file in the path that is returned.
+    """
+    result_filename = get_tmp_file_name()
```

Comment:
Would Python tempfile module with suffix parameter work just as well here?

Reply:
I considered that. Using tempfile would not be a good fit here, because the tempfiles get destroyed when they are closed, so basically there would be a massive context manager block or something like that.

Reply:
Tempfile is not a good fit here, because the file gets destroyed once the handle is closed.